atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTableCreator.java

```diff
@@ -0,0 +1,134 @@
+
+    CassandraTableCreator(CassandraClientPool clientPool, CassandraKeyValueServiceConfig config) {
+        this.clientPool = clientPool;
+        this.config = config;
+    }
+
+    void createTables(Map<TableReference, byte[]> tableRefToMetadata) {
+        try {
+            clientPool.runWithRetry(client -> {
+                for (Map.Entry<TableReference, byte[]> entry : tableRefToMetadata.entrySet()) {
+                    CassandraKeyValueServices.runWithWaitingForSchemas(
+                            () -> createTable(entry.getKey(), entry.getValue(), client), config, client,
+                            "adding the column family for table " + entry.getKey() + " in a call to create tables");
+                }
+                return null;
+            });
+        } catch (Exception e) {
+            throw Throwables.unwrapAndThrowAtlasDbDependencyException(e);
+        }
+    }
+
+    private void createTable(TableReference tableRef, byte[] metadata, CassandraClient client) throws TException {
+        CqlQuery query = constructQuery(tableRef, metadata);
+        client.execute_cql3_query(query, Compression.NONE, CassandraKeyValueServiceImpl.WRITE_CONSISTENCY);
+    }
+
+    private CqlQuery constructQuery(TableReference tableRef, byte[] metadata) {
+        TableMetadata tableMetadata = CassandraKeyValueServices.getMetadataOrDefaultToGeneric(metadata);
+        boolean appendHeavyReadLight = tableMetadata.isAppendHeavyAndReadLight();
+        String keyspace = wrapInQuotes(config.getKeyspaceOrThrow());
+        String internalTableName = wrapInQuotes(CassandraKeyValueServiceImpl.internalTableName(tableRef));
+
+        String query = SchemaBuilder.createTable(keyspace, internalTableName)
+                .ifNotExists()
+                .addPartitionKey(ROW, DataType.blob())
+                .addClusteringColumn(COLUMN, DataType.blob())
+                .addClusteringColumn(TIMESTAMP, DataType.bigint())
+                .addColumn(VALUE, DataType.blob())
+                .withOptions()
+                .bloomFilterFPChance(falsePositive(tableMetadata.hasNegativeLookups(), appendHeavyReadLight))
+                .caching(SchemaBuilder.Caching.KEYS_ONLY)
+                .comment("")
+                .compactionOptions(getCompaction(appendHeavyReadLight))
+                .compactStorage()
+                .compressionOptions(getCompression(tableMetadata.getExplicitCompressionBlockSizeKB()))
+                .dcLocalReadRepairChance(0.1)
+                .gcGraceSeconds(config.gcGraceSeconds())
+                .minIndexInterval(128)
+                .maxIndexInterval(2048)
+                .populateIOCacheOnFlush(tableMetadata.getCachePriority() == CachePriority.HOTTEST)
+                .speculativeRetry(SchemaBuilder.noSpeculativeRetry())
+                .clusteringOrder(COLUMN, SchemaBuilder.Direction.ASC)
+                .clusteringOrder(TIMESTAMP, SchemaBuilder.Direction.ASC)
+                .buildInternal();
```

Comment:
Also the following things are explicitly set to defaults in the thrift implementation that I was unable to set here. Is this something that we must do @tpetracca ?
cf.setTriggers(ImmutableList.of());
cf.setCells_per_row_to_cache("0");
cf.setColumn_metadata(ImmutableList.of());
cf.setKey_validation_class("org.apache.cassandra.db.marshal.BytesType");
cf.setDefault_validation_class("org.apache.cassandra.db.marshal.BytesType");

Reply:
You should be all good:
- triggers are their own kind of statement `CREATE TRIGGER` so there's no concept of it in a createTable statement in CQL
- we're not caching rows anyway
- the bottom 3 are all covered in a different way by the top section here (the row, column1, column2, value names and their types)

Reply:
Thanks for confirming, I assumed that was the case since the comparator worked that way